extensions/interactions/NumberWithUnits/directives/NumberWithUnitsValidationServiceSpec.js

```diff
@@ -0,0 +1,205 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+describe('NumberWithUnitsValidationService', function() {
+  var validatorService, WARNING_TYPES;
+
+  var currentState;
+  var answerGroups, goodDefaultOutcome;
+  var equalsTwoRule, equalsTwoByThreeRule, equivalentToTwoThousandRule,
+    equivalentToTwoByThreeRule, equivalentToTwoRule;
+  var createNumberWithUnitsDict;
+  var oof, agof, rof;
+  beforeEach(function() {
+    module('oppia');
+  });
+
+  beforeEach(inject(function($injector) {
+    validatorService = $injector.get('NumberWithUnitsValidationService');
+    oof = $injector.get('OutcomeObjectFactory');
+    agof = $injector.get('AnswerGroupObjectFactory');
+    rof = $injector.get('RuleObjectFactory');
+    WARNING_TYPES = $injector.get('WARNING_TYPES');
+
+    var createFractionDict = function(
+        isNegative, wholeNumber, numerator, denominator) {
+      return {
+        isNegative: isNegative,
+        wholeNumber: wholeNumber,
+        numerator: numerator,
+        denominator: denominator
+      };
+    };
+
+    var createUnitsDict = function(unitDict) {
+      return {
+        units: unitDict
+      };
+    };
+
+    var createNumberWithUnitsDict = function(
+        type, real, fractionDict, unitDict) {
+      return {
+        type: type,
+        real: real,
+        fraction: fractionDict,
+        units: unitDict
```

Comment:
The value to this key should be a unitDict which represents unit directly. Right now it is another dict whose key is unit and value is the actual unitDict.